### ISSUE TITLE
Specify the scope of CPC votes during application and incubation

### ISSUE THREAD

**Issue by tobie** [openjs-foundation/cross-project-council#296]
During the silent period, the CPC votes to accept the application of a project to join the OpenJSF. One the project is ready to leave incubation, the CPC votes again to allow the project to graduate. Additionally, the CPC can request that a project withdraw at any point of the incubation phase without having to provide justification for it.

As I understand it, the point of the silent period is to make sure there's a fit between a project and the foundation and to have room to reject an application if that's not the case, without having to publicly hurt people's feelings, the image of a project, or that of the corporations backing it.

Once this fit is established and the project's application has been accepted, it feels that the reasons to request withdrawal should be scoped and the request motivated. Requesting a withdrawal should be limited to projects breaking rules or having stalled on their path to graduation and not making concrete steps to get back on track.

Similarly, once a project is ready for graduation, the CPC's votes should be limited to decide whether the project meets graduation criteria. Other concerns, such as fit, should have been brought up during the initial vote.

Joining a foundation is a large investment for a project and the people behind it. It would be unfair to allow a project to invest important resources into this process only to have the CPC refuse graduation or request withdrawal on motives the project has no control over and which weren't specified in the graduation criteria or onboarding checklist.

**Comment by mhdawson** [comment]
> Once this fit is established and the project's application has been accepted, it feels that the reasons to request withdrawal should be scoped and the request motivated. Requesting a withdrawal should be limited to projects breaking rules or having stalled on their path to graduation and not making concrete steps to get back on track.

I'm not quite sure I agree with this.  I agree the quiet period is to make sure the application does not proceed unless the CPC members think there is a good change of success.  Having said that, the CPC is a smaller number of people and additional information/feedback may come to light once the application is public and we want that public feedback to influence the decision...

My opinion is that the quiet period is an important part to avoiding the situation which nobody wants (application being refused after being submitted) but that it should not be a "done deal" after the submission with only "bad actions" on the part of the project being a valid reason for graduation.  If more information/further discussion means that it now understood that the project is not a good fit the CPC should make that hard decision and be trusted to do so.

**Comment by tobie** [comment]
Thank you, @mhdawson, for your comment. That helps me better understand how folks on the CPC perceived the process and what their perspective is.

Could the CPC maybe define "good fit" in a way a project can self-assess before kick-starting the procedure? Right now, this seems really subjective and completely dependant on whoever sits on the CPC. Are their existing guidelines to frame that conversation?

**Comment by dmethvin** [comment]
In most cases I suspect we'd graduate any project that met the acceptance criteria for their stage.  I'm  hesitant to say that the CPC would be absolutely _required_ to graduate a project that met the listed criteria if there were other things that raised red flags. 

I do think the up-front work is the most important. The CPC shouldn't be accepting projects unless we think they can graduate. I'll make up a scenario below that represents a worrisome situation:

> "I have gotten permission from my company to open-source a JavaScript project we've had for a few years. It's really cool but doesn't have users outside our company. The company switched to a new tech stack internally and doesn't use this project anymore, I've been the sole maintainer for the last year as the company transitioned to the new stack. They aren't interested in letting me work on the project much during work hours but I plan to work on it in my spare time. It would be great to be in the OpenJS Foundation so I could get some help building this project."

In summary: older technology, few current users, one part-time maintainer, no support from the company that's open-sourcing the code.

I would not call this a good fit. However, others on the CPC might argue that the project incorporates some great ideas and just hasn't had enough mentorship and exposure to get the users and contributors it needs. The chances are that it would require a lot of work on someone's part, so I would only want to admit a project if we had specific people willing to do that work.

**Comment by MylesBorins** [comment]
When @jorydotcom and I drafted the process for the silent period it was not the intent (afaik) to have a vote during the silent period. The silent period was meant to derive rough-consensus and ensure that any large obvious blocking concerns could be handled quietly and respectfully. In a way the process doesn't really start until a project is in "incubation".

As with many other votes in our foundation I would imagine that by the time we get to the voting process the vote is primarily being done as a formality. waiting until a vote to derail the process would be imho quite bad faith, and something that should have been sorted out earlier.

What types of documented process would help alleviate some of your concerns?
